opba-api-security/src/main/java/de/adorsys/opba/api/security/internal/service/TokenBasedAuthService.java

```diff
@@ -39,6 +39,20 @@ public String generateToken(String subject) {
         return signedJWT.serialize();
     }
 
+    @SneakyThrows
+    public String generateRedirectToAspspToken(String subject) {
```

Comment:
Why duplicating `generateToken` function from above? You need to pass Duration as an argument

```
public generateToken(...) {
doGenerateToken(subject, tppTokenProperties.getTokenValidityDuration());
}

public generateRedirectToAspspToken(...) {
doGenerateToken(subject, tppTokenProperties.getRedirectTokenValidityDuration());
}

private String doGenerateToken(String subject, Duration tokenLifetime) {
        ZonedDateTime currentTime = ZonedDateTime.now(ZoneOffset.UTC);
        JWTClaimsSet claims = new JWTClaimsSet.Builder()
                                      .expirationTime(Date.from(currentTime.plus(tokenLifetime).toInstant()))
                                      .issueTime(Date.from(currentTime.toInstant()))
                                      .subject(String.valueOf(subject))
                                      .build();
        SignedJWT signedJWT = new SignedJWT(jwsHeader, claims);
        signedJWT.sign(jwsSigner);
        return signedJWT.serialize();
}
```

Reply:
done